### ISSUE TITLE
Possible to delete a stubborn tag in Firefox without deleting all bookmarks?

### ISSUE THREAD

**Issue by Gitoffthelawn** [aaFn/Bookmark-search-plus-2#231]
This is a bit off-topic, but @aaFn, I figure you're likely an expert on this topic.

I have a couple stubborn bookmark tags that cannot be removed via the native Firefox Bookmarks Library.

This happens fairly regularly in Firefox (even without BSP2 installed, so it's not a BSP2 issue).  The only solution I've found is to export all bookmarks, delete `places.sqlite`, and then import the bookmarks back into Firefox.  The catch is that this loses all the favicons, and I find favicons to be very useful.  With tens of thousands of bookmarks, visiting each bookmark to regenerate the favicon (for within Firefox's native UI, not BSP2) is too time consuming.

I've been going through all the Web Extension APIs, and I haven't been able to find an API to allow me to delete a tag.

Is there a way to delete a tag via the Web Extension API or via the Firefox browser console?

TIA.

**Comment by aaFn** [comment]
Hello @Gitoffthelawn , as you can see in https://bugzilla.mozilla.org/show_bug.cgi?id=1225916, Web Extensions do not have a way to manipulate tags.

And as you can see in https://bugzilla.mozilla.org/show_bug.cgi?id=1636755, tags are not associated with bookmarks, but with URLs = it looks like they exist independently from bookmarks then, managed in `PlacesTransactions` (https://firefox-source-docs.mozilla.org/browser/places/PlacesTransactions.html).

I do not know if there is a way to managed stored tags directly in Firefox UI through `PlacesTransactions` or anything else, but I hope this helps in your quest.

**Comment by Gitoffthelawn** [comment]
Hi @aaFn.  Thank you for the information.  I'm happy you were able to confirm that there is no help from the Web Extensions API.  Well, not happy about the lack of functionality, but happy I didn't miss anything. ;)

Your mentioning of tags being tied to URLs and not to bookmarks is quite interesting.  I noticed something odd along those lines during my research, as that's what I thought I discovered too.  I conjectured it was most likely I wasn't understanding something correctly, but with your helpful and encouraging words, I'm going to take another look and confirm what I thought I originally saw.  I'm trying to figure out *why* Mozilla would want tags tied to URLs instead of bookmarks.  Any thoughts on that?

Your comments regarding `PlacesTransactions` may be quite helpful.  I'm not very familiar with that part of Firefox's source code, so it's another place to look.  Documentation of their code is slim (and their code is not well-commented), so I'm open to any thoughts from you or anyone else stumbling upon our words.
